rpcs3/rpcs3qt/infinity_dialog.cpp

```diff
@@ -407,46 +407,93 @@ figure_creator_dialog::figure_creator_dialog(QWidget* parent, u8 slot)
 {
 	setWindowTitle(tr("Figure Creator"));
 	setObjectName("figure_creator");
-	setMinimumSize(QSize(500, 150));
+	setMinimumSize(QSize(500, 200));
 
 	QVBoxLayout* vbox_panel = new QVBoxLayout();
 
+	// Add series filter buttons
+	QHBoxLayout* hbox_filters = new QHBoxLayout();
+	QLabel* filter_label = new QLabel(tr("Filter by Series:"));
+	QPushButton* btn_all = new QPushButton(tr("All"));
+	QPushButton* btn_series1 = new QPushButton(tr("1.0"));
+	QPushButton* btn_series2 = new QPushButton(tr("2.0"));
+	QPushButton* btn_series3 = new QPushButton(tr("3.0"));
+
+	// Style the active filter button
+	btn_all->setCheckable(true);
+	btn_series1->setCheckable(true);
+	btn_series2->setCheckable(true);
+	btn_series3->setCheckable(true);
+	btn_all->setChecked(true); // Default to "All"
+
+	hbox_filters->addWidget(filter_label);
+	hbox_filters->addWidget(btn_all);
+	hbox_filters->addWidget(btn_series1);
+	hbox_filters->addWidget(btn_series2);
+	hbox_filters->addWidget(btn_series3);
+	hbox_filters->addStretch();
+
+	vbox_panel->addLayout(hbox_filters);
+
 	QComboBox* combo_figlist = new QComboBox();
 	QStringList filterlist;
 	u32 first_entry = 0;
 
-	for (const auto& [figure, entry] : list_figures)
-	{
-		// Only display entry if it is a piece appropriate for the slot
-		if ((slot == 0 &&
-				((figure > 0x1E8480 && figure < 0x2DC6BF) || (figure > 0x3D0900 && figure < 0x4C4B3F))) ||
-			((slot == 1 || slot == 2) && (figure > 0x3D0900 && figure < 0x4C4B3F)) ||
-			((slot == 3 || slot == 6) && figure < 0x1E847F) ||
-			((slot == 4 || slot == 5 || slot == 7 || slot == 8) &&
-				(figure > 0x2DC6C0 && figure < 0x3D08FF)))
+	// Lambda to populate the combo box based on series filter
+	auto populate_combo = [&](int series_filter) {
```

Comment:
This needs to happen with =, not &, otherwise the pointers will be out of scope...
```suggestion
	auto populate_combo = [=](int series_filter)
	{
```